packages/airnode-node/src/workers/cloud-platforms/gcp.test.ts

```diff
@@ -0,0 +1,67 @@
+const requestMock = jest.fn();
+const getIdTokenClientMock = jest.fn().mockImplementation(() => ({
+  request: requestMock,
+}));
+jest.mock('google-auth-library', () => ({
+  GoogleAuth: jest.fn().mockImplementation(() => ({
+    getIdTokenClient: getIdTokenClientMock,
+  })),
+}));
+
+import * as gcp from './gcp';
+import * as fixtures from '../../../test/fixtures';
+import { WorkerFunctionName } from '../../types';
+
+describe('spawn', () => {
+  it('derives the function URL, authenticates, invokes and returns the response', async () => {
+    requestMock.mockImplementationOnce(() => ({ data: { value: 7777 } }));
+    const workerOpts = fixtures.buildWorkerOptions({
+      cloudProvider: { name: 'gcp', region: 'us-east1', projectId: 'projectId123' },
+    });
+    const parameters = {
+      ...workerOpts,
+      functionName: 'some-function' as WorkerFunctionName,
+      payload: { from: 'ETH', to: 'USD' },
+    };
+    const url = 'https://us-east1-projectId123.cloudfunctions.net/airnode-19255a4-test-some-function';
+
+    const res = await gcp.spawn(parameters);
+    expect(res).toEqual({ value: 7777 });
+
+    expect(getIdTokenClientMock).toHaveBeenCalledTimes(1);
+    expect(getIdTokenClientMock).toHaveBeenCalledWith(url);
+
+    expect(requestMock).toHaveBeenCalledTimes(1);
+    expect(requestMock).toHaveBeenCalledWith({
+      url,
+      method: 'POST',
+      data: parameters.payload,
+    });
+  });
+
+  it('throws an error if the cloud function returns an error', async () => {
+    expect.assertions(5);
```

Comment:
I don't like asserting the number of assertions.

Reply:
That's actually a suggestion from the [Jest documentation](https://jestjs.io/docs/expect#expectassertionsnumber) when you're testing async functions.

Reply:
But that is for a different use case - it ensures that your callback assertions were called. You don't need to add it just because you write async tests